### PR TITLE
#LESQ-1000: Newsletter signup check box

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 2016

- Adds form and controller to the onboarding page form
- Default selects newsletterSignUp to be true 
- Submits the true or false value to the data from object

## Issue(s)

AC for ticket has slightly changed with Hubspot intergration being extrapolated into separate ticket and Manual input of school not created as of yet

Fixes #LESQ-1000

## How to test

1. Go to https://deploy-preview-2683--oak-web-application.netlify.thenational.academy/onboarding
2. You should see onboarding form now has sign up to newsletter checkbox defaulted to true

## Screenshots

How it should now look:
Screenshot 2024-08-09 at 12 04 52

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherViews/Onboarding/Onboarding.view.tsx [hunk 1]
```diff
@@ -23,17 +24,22 @@ const onboardingFormSchema = z.object({
     })
     .min(1, "Select school"),
   schoolName: z.string().optional(),
+  newsletterSignUp: z.boolean().optional(),
```
Comment (njblore) [inline src/components/TeacherViews/Onboarding/Onboarding.view.tsx:27]: don't think this should be optional

**Review by jack-skerrett-oak — APPROVED** [review]
It is a checkbox that doesn't do a whole lot just yet, great job :D

File: src/components/TeacherViews/Onboarding/Onboarding.test.tsx [hunk 2]
```diff
@@ -40,6 +41,28 @@ describe("Onboarding view", () => {
...
+  it("should render the Controller component and handle checkbox change", async () => {
+    const { getByRole } = renderWithProviders()(<OnboardingView />);
+
+    const checkbox = getByRole("checkbox", {
+      name: /Sign up to receive helpful content via email. Unsubscribe at any time./i,
+    });
+    expect(checkbox).toBeChecked();
+
+    await act(async () => {
+      fireEvent.click(checkbox);
```
Comment (njblore) [inline src/components/TeacherViews/Onboarding/Onboarding.test.tsx:61]: pretty sure fireEvent wraps in `act` internally - were you seeing warnings without it?

Comment (k-huggs) [reply src/components/TeacherViews/Onboarding/Onboarding.test.tsx:61]: Yeah was getting loads of console warnings so chucked this around it, happy to remove / ignore them?

Comment (njblore) [reply src/components/TeacherViews/Onboarding/Onboarding.test.tsx:61]: nah definitely don't ignore them, I'd maybe try quickly refactoring to the `await`/`findBy` style of the other tests in here just for clarity, but if it is still having issues this is fine

**Review by njblore — APPROVED** [review]

File: src/components/TeacherViews/Onboarding/Onboarding.test.tsx (hunk 2 above)
Comment (k-huggs) [reply src/components/TeacherViews/Onboarding/Onboarding.test.tsx]: Refactored ✅

**Pull request merged**